web-api/tests/store_user_history.rs

```diff
@@ -0,0 +1,110 @@
+// Copyright 2023 Xayn AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use std::collections::HashSet;
+
+use reqwest::StatusCode;
+use serde::Deserialize;
+use serde_json::json;
+use toml::toml;
+use xayn_integration_tests::{send_assert, send_assert_json, test_two_apps};
+use xayn_web_api::{Ingestion, Personalization};
+
+#[derive(Deserialize)]
+struct PersonalizedDocumentData {
+    id: String,
+}
+
+#[derive(Deserialize)]
+struct PersonalizedDocumentsResponse {
+    documents: Vec<PersonalizedDocumentData>,
+}
+
+async fn store_user_history(enabled: bool) {
+    test_two_apps::<Ingestion, Personalization, _>(
+        |_| {},
+        |config| {
+            config.extend(toml! {
+                [personalization]
+                store_user_history = enabled
+            });
```

Comment:
While this works in this case it is bug prone compared to the `set_config_option` macro.

:wrench:  The difference is that `extend` is based on the `Map` type while the `set_config_option` macro is based on the full path per option.

So if you e.g. `extend`  with a hypothetical  `[storage.postgres] max_connections=2` then it will replace the _all_ storage entries with just the `max_connections` setting including the url.

Through this is unrelated to the CI failure.

Reply:
i didn't expect that, i'll fix it before merging.